src/parser/xasl_generation.c

```diff
@@ -702,7 +706,13 @@ pt_make_connect_by_proc (PARSER_CONTEXT * parser, PT_NODE * select_node, XASL_NO
       PT_NODE *save_where, *save_from;
 
       save_where = select_node->info.query.q.select.where;
-      select_node->info.query.q.select.where = select_node->info.query.q.select.connect_by;
+      where = select_node->info.query.q.select.connect_by;
+
+      /* Separate the predicate related to the HQ. */
+      /* pred_with_HQ is appended to if_pred instead of data filter. */
+      pt_split_pred_for_HQ (parser, where, &pred_without_HQ, &pred_with_HQ);
+
+      select_node->info.query.q.select.where = pred_with_HQ ? pred_without_HQ : where;
```

Comment:
If pred_with_HQ is not null, pred_without_HQ should be null. 
If so, does it have the same meaning as the code below?
`select_node->info.query.q.select.where = pred_with_HQ ? NULL : where`

Reply:
No. If pred_with_HQ is null, it means there is no change from the existing where.
If pred_with_HQ is not null, pred_without_HQ is null or not null.

Reply:
O, I didn't see the for loop properly in pt_split_pred_for_HQ()
thank you~